bundles/org.openhab.binding.etherrain/src/main/java/org/openhab/binding/etherrain/internal/handler/EtherRainHandler.java

```diff
@@ -0,0 +1,239 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.etherrain.internal.handler;
+
+import java.io.IOException;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.etherrain.internal.EtherRainBindingConstants;
+import org.openhab.binding.etherrain.internal.EtherRainException;
+import org.openhab.binding.etherrain.internal.api.EtherRainCommunication;
+import org.openhab.binding.etherrain.internal.api.EtherRainStatusResponse;
+import org.openhab.binding.etherrain.internal.config.EtherRainConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EtherRainHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Joe Inkenbrandt - Initial contribution
+ */
+@NonNullByDefault
+public class EtherRainHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(EtherRainHandler.class);
+
+    private @Nullable EtherRainCommunication device = null;
+    private boolean connected = false;
+    private @NonNullByDefault({}) EtherRainConfiguration config = null;
+
+    private @Nullable ScheduledFuture<?> updateJob = null;
+
+    private final HttpClient httpClient;
+
+    /*
+     * Constructor class. Only call the parent constructor
+     */
+    public EtherRainHandler(Thing thing, HttpClient httpClient) {
+        super(thing);
+        this.httpClient = httpClient;
+        this.updateJob = null;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command.toString().equals("REFRESH")) {
+            updateBridge();
```

Comment:
Because the thread calling `handleCommand` is usually a UI thread, it is expected that it will return quickly. This method makes an http request which can be slow. So I suggest that you call this method asynchronously to make sure you don't hog the UI thread.
```suggestion
            scheduler.execute(this::updateBridge);
```